packages/react-app/src/transaction/CreateSpeedUpTransaction/CreateSpeedUpTransaction.tsx

```diff
@@ -0,0 +1,204 @@
+      <FormFieldWrapper>
+        <FormLabel>Gas price</FormLabel>
+        <Box className={classes.inputField}>
+          <Box className={classes.gasPriceSliderBox}>
+            <Slider
+              aria-labelledby="discrete-slider"
+              classes={{ markLabel: classes.gasPriceMarkLabel }}
+              className={classes.gasPriceSlider}
+              defaultValue={minGasPriceNumber}
+              marks={[
+                { value: minGasPriceNumber, label: 'Normal' },
+                { value: maxGasPriceNumber, label: 'Urgent' },
+              ]}
+              max={maxGasPriceNumber}
+              min={minGasPriceNumber}
+              step={0.01}
+              valueLabelDisplay="auto"
+              valueLabelFormat={(value): string => value.toFixed(2)}
+              getAriaValueText={(): string => `${gasPrice.toFixed(2)} ${txGasPriceUnit.toString()}`}
+              onChange={(event, value): void => setGasPrice(value as number)}
+            />
+          </Box>
+          <Box className={classes.gasPriceHelpBox}>
+            <FormHelperText className={classes.gasPriceHelp}>
+              {gasPrice.toFixed(2)} {txGasPriceUnit.toString()}
+            </FormHelperText>
+          </Box>
+        </Box>
+      </FormFieldWrapper>
+      <FormFieldWrapper>
+        <FormLabel>Password</FormLabel>
+        <PasswordInput error={passwordError} onChange={setPassword} />
+      </FormFieldWrapper>
+      <FormFieldWrapper>
+        <FormLabel />
+        <ButtonGroup style={{ width: '100%' }}>
+          <Button label="Cancel" onClick={goBack} />
+          <Button
+            label="Sign Transaction"
+            disabled={password.length === 0}
+            primary={true}
+            onClick={() => onSignTransaction({ ...transaction, gasPrice: new Wei(gasPrice, txGasPriceUnit).number })}
+          />
+        </ButtonGroup>
+      </FormFieldWrapper>
+    </Page>
+  );
+};
+
+export default connect<{}, DispatchProps, OwnProps, IState>(
+  (state, ownProps) => {
+    const { blockchain, from } = ownProps.transaction;
+
+    const account = accounts.selectors.findAccountByAddress(state, from, blockchain);
+
+    return { accountId: account?.id };
+  },
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  (dispatch: any) => ({
+    checkGlobalKey(password) {
+      return dispatch(accounts.actions.verifyGlobalKey(password));
+    },
+    getTopFee(blockchain) {
+      return dispatch(transaction.actions.estimateFee(blockchain, 128, 'avgTop'));
+    },
+    goBack() {
+      dispatch(screen.actions.goBack());
+    },
+    async signTransaction(tx, password, accountId) {
+      if (accountId == null || tx.to == null) {
+        return;
+      }
+
+      const signed = await dispatch(
+        transaction.actions.signTransaction(
+          accountId,
+          tx.blockchain,
+          tx.from,
+          password,
+          tx.to,
+          21000,
```

Comment:
Should be original `tx.gas`. 

We change only gas price here, every other field stays the same